File: C/Colleen.c
#include <stdio.h>

/*
  this is a fixed-point program: its output is identical to its source code
*/

static void say(const char *s)
{
	printf(s, 10, 9, 34, s);
}

int main(void)
{
	const char *s = "#include <stdio.h>%1$c%1$c/*%1$c  this is a fixed-point program: its output is identical to its source code%1$c*/%1$c%1$cstatic void say(const char *s)%1$c{%1$c%2$cprintf(s, 10, 9, 34, s);%1$c}%1$c%1$cint main(void)%1$c{%1$c%2$cconst char *s = %3$c%4$s%3$c;%1$c%2$c/*%1$c%2$c  above is a string containing the source code%1$c%2$c*/%1$C%1$c%2$csay(s);%1$c%1$c%2$creturn 0;%1$c}%1$c";
	/*
	  above is a string containing the source code
	*/

	say(s);

	return 0;
}

File: Python/Colleen.py
"""
	this is a fixed-point program: its output is identical to its source code
"""

def say(s: str) -> None:
	print(s.format(lf=chr(10), ht=chr(9), quote=chr(39), format=s))

def main():
	s = '"""{lf}{ht}this is a fixed-point program: its output is identical to its source code{lf}"""{lf}{lf}def say(s: str) -> None:{lf}{ht}print(s.format(lf=chr(10), ht=chr(9), quote=chr(39), format=s)){lf}{lf}def main():{lf}{ht}s = {quote}{format}{quote}{lf}{ht}"""{lf}{ht}{ht}above is a string containing the source code{lf}{ht}"""{lf}{lf}{ht}say(s){lf}{lf}if __name__ == "__main__":{lf}{ht}main()'
	"""
		above is a string containing the source code
	"""

	say(s)

if __name__ == "__main__":
	main()
